code/modules/atmospherics/machinery/components/binary_devices/volume_pump.dm

```diff
@@ -64,44 +62,57 @@
 		cut_overlay(overclock_overlay)
 
 /obj/machinery/atmospherics/components/binary/volume_pump/process_atmos()
-	if(!on || !is_operational)
+	if(!can_process())
 		return
+	do_transfer()
 
-	var/datum/gas_mixture/air1 = airs[1]
-	var/datum/gas_mixture/air2 = airs[2]
+/// -------------------------------
+/// Helper procs
+/// -------------------------------
 
-// Pump mechanism just won't do anything if the pressure is too high/too low unless you overclock it.
+/obj/machinery/atmospherics/components/binary/volume_pump/proc/can_process()
+	if(!on || !is_operational)
+		return FALSE
+	var/datum/gas_mixture/input_air = airs[1]
+	var/datum/gas_mixture/output_air = airs[2]
+	if(!input_air || !output_air)
+		return FALSE
 
-	var/input_starting_pressure = air1.return_pressure()
-	var/output_starting_pressure = air2.return_pressure()
+	var/input_starting_pressure = input_air.return_pressure()
+	var/output_starting_pressure = output_air.return_pressure()
 
-	if((input_starting_pressure < 0.01) || ((output_starting_pressure > 9000)) && !overclocked)
-		return
+	if((input_starting_pressure < VOLUME_PUMP_MIN_INPUT_PRESSURE) || (output_starting_pressure > VOLUME_PUMP_MAX_OUTPUT_PRESSURE && !overclocked))
+		return FALSE
 
-///Monkestation edit begin
-/*  Volume pumps can now pump infinitely while overclocked -
-	if(overclocked && (output_starting_pressure-input_starting_pressure > 1000))//Overclocked pumps can only force gas a certain amount.
-		return
-*/
-///Monkestation edit end
+	return TRUE
 
-	var/transfer_ratio = transfer_rate / air1.volume
+/obj/machinery/atmospherics/components/binary/volume_pump/proc/do_transfer()
+	var/datum/gas_mixture/input_air = airs[1]
+	var/datum/gas_mixture/output_air = airs[2]
 
-	var/datum/gas_mixture/removed = air1.remove_ratio(transfer_ratio)
+	var/transfer_ratio = transfer_rate / input_air.volume
+	var/datum/gas_mixture/removed = input_air.remove_ratio(transfer_ratio)
 
 	if(!removed.total_moles())
 		return
 
-	if(overclocked)//Some of the gas from the mixture leaks to the environment when overclocked
-		var/turf/open/T = loc
-		if(istype(T))
-			var/datum/gas_mixture/leaked = removed.remove_ratio(VOLUME_PUMP_LEAK_AMOUNT)
-			T.assume_air(leaked)
-
-	air2.merge(removed)
+	// only leak gas when overclocked
+	if(overclocked)
+		handle_overclock_leak(removed)
 
+	// push remaining gas into output
+	output_air.merge(removed)
 	update_parents()
 
+/// Leak helper
+/obj/machinery/atmospherics/components/binary/volume_pump/proc/handle_overclock_leak(datum/gas_mixture/removed)
+	var/turf/current_turf = loc
+	if(!isclosedturf(current_turf) && isopenturf(current_turf))
```

Comment:
```suggestion
	if(isopenturf(current_turf))
```
if it's an open turf, then it's not a closed turf
sorry, should have been more clear in my review

i know this is a nitpick, but like, it's atmos code, better safe than sorry when it comes to performance

i'll merge right after this is fixed